src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -111,12 +89,12 @@ internal void AddProvidersFromAppSettings()
                 if (workerDirectorySection.Value != null)
                 {
                     _workerDescriptionDictionary.Remove(languageSection.Key);
-                    AddProvider(workerDirectorySection.Value);
+                    AddProvider(workerDirectorySection.Value, workerConfigurationInfo);
                 }
             }
         }
 
-        internal void AddProvider(string workerDir)
+        internal void AddProvider(string workerDir, WorkerConfigurationInfo workerConfigurationInfo)
```

Comment:
nit: if you only need to use one property inside of a method, you can be more specific here. In this case, if you only need the `WorkersRootDirPath`, this parameter could change to a string `workersRootDirPath`. It makes things more clear. Although my other comment about making this a bool may be even more clear.